### PR TITLE
GPC-351 Delete Event Function

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: terraform/modules/gro_ingestion_service/delete_event.tf [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      "kms:ReEncryptTo"
+    ]
+    effect = "Allow"
+    resources = [
+      aws_kms_key.gro_ingestion.arn
+    ]
+  }
+
+  statement {
+    sid = "DynamoDBGetItemAccess"
```
Comment (oskarwilliams378) [inline terraform/modules/gro_ingestion_service/delete_event.tf:24]: this may be wrong

File: terraform/modules/gro_ingestion_service/lambdas/test/functions/dbItem.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+export const dbItem = {
```
Comment (oskarwilliams378) [inline terraform/modules/gro_ingestion_service/lambdas/test/functions/dbItem.ts:1]: Should we add this to a mocks or consts folder?

File: terraform/modules/gro_ingestion_service/lambdas/test/functions/deleteEvent.function.spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+import { DynamoDBClient, dynamoDbSendFn } from "../__mocks__/@aws-sdk/client-dynamodb"
+
+const db = new DynamoDBClient()
+
+describe("Unit test for delete event handler", function () {
+    test("verifies successful response", async () => {
+        dynamoDbSendFn.mockReturnValueOnce(Promise.resolve({ Item: dbItem }))
+
+        const event: EventRequest = {
```
Comment (oskarwilliams378) [inline terraform/modules/gro_ingestion_service/lambdas/test/functions/deleteEvent.function.spec.ts:16]: we're using this a few times, should this also be a const?

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    try {
+        const command = new DeleteItemCommand(params)
+        const result = await dynamo.send(command)
+
+        let logParams = {
+            hash: event.id,
+            RegistrationId: "",
+            EventTime: "",
+        }
+        if (result.Attributes) {
```
Comment (oskarwilliams378) [inline terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts:30]: How would we get to the case where there are no Attributes here?

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            logParams = {
+                ...logParams,
+                RegistrationId: eventRecord?.RegistrationId,
+                EventTime: eventRecord?.EventTime,
+            }
+        }
+        console.log("Successfully deleted event", logParams)
+        return {
+            id: event.id,
+            statusCode: 204,
```
Comment (oskarwilliams378) [inline terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts:41]: as we are returning an ID would this be a 200 rather than 204?

**Review by oskarwilliams378 — CHANGES_REQUESTED** [review]

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts (hunk 4 above)
Comment (Mjwillis) [reply terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts:30]: It was just a edge case where the attributes where not returned but probs not possible

File: terraform/modules/gro_ingestion_service/lambdas/test/functions/dbItem.ts (hunk 2 above)
Comment (Mjwillis) [reply terraform/modules/gro_ingestion_service/lambdas/test/functions/dbItem.ts:1]: yes

File: terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts (hunk 5 above)
Comment (Mjwillis) [reply terraform/modules/gro_ingestion_service/lambdas/src/functions/deleteEvent.function.ts:41]: yes

File: terraform/modules/gro_ingestion_service/lambdas/test/functions/deleteEvent.function.spec.ts (hunk 3 above)
Comment (Mjwillis) [reply terraform/modules/gro_ingestion_service/lambdas/test/functions/deleteEvent.function.spec.ts:16]: Actually its different

Comment (oskarwilliams378) [reply terraform/modules/gro_ingestion_service/lambdas/test/functions/deleteEvent.function.spec.ts:15]: Here and in enrichEvent it looks to be the below, whats the difference?
```
        const event: EventRequest = {
            id: "hash1",
        }
```

File: terraform/modules/gro_ingestion_service/lambdas/test/const/dbItem.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        S: "Forename",
+    },
+    Surname: {
+        S: "Surname",
+    },
+    Sex: {
+        S: "Male",
+    },
+    MaidenSurname: {
+        S: null,
```
Comment (oskarwilliams378) [inline terraform/modules/gro_ingestion_service/lambdas/test/const/dbItem.ts:24]: I think these should be empty string not null, as we will want to make sure the record is returned with that and is stored with that

**Review by oskarwilliams378 — CHANGES_REQUESTED** [review]
A couple comments left :)
